### ISSUE TITLE
ansible-lint type checks don't account for jinja2 expressions that change data type

### ISSUE THREAD

**Issue by Drugoy** [ansible/ansible-lint#4043]
##### Summary

ansible-lint type checks don't account for jinja2 expressions that change data type

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 24.2.0 using ansible-core:2.15.5 ansible-compat:4.1.11 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Create a role with the following files:

```yaml
defaults/main.yml
---
mydict:
  a: a
  b: b


meta/argument_specs.yml
---
argument_specs:
  main:
    options:
      myvar:
        type: str
        required: true
        choices: "{{ mydict.keys() | list }}"
```

##### Desired Behavior

ansible-lint should understand that jinja2 expression that is used in `choices` actually yields an array

##### Actual Behavior

schema[role-arg-spec]: $.argument_specs.main.options.myvar.choices '{{ mydict.keys() | list }}' is not of type 'array'.

**Comment by ssbarnea** [comment]
That seems like another schema error, as the schema should allow jinja templates for this field too. Please make a PR to fix it.
